src/V3Sched.cpp

```diff
@@ -0,0 +1,964 @@
+AstCFunc* createInitial(AstNetlist* netlistp, const LogicClasses& logicClasses) {
+    AstCFunc* const funcp = makeTopFunction(netlistp, "_eval_initial", /* slow: */ true);
+    orderSequentially(funcp, logicClasses.m_initial);
+    return funcp;  // Not splitting yet as it is not final
+}
+
+void createFinal(AstNetlist* netlistp, const LogicClasses& logicClasses) {
+    AstCFunc* const funcp = makeTopFunction(netlistp, "_final", /* slow: */ true);
+    orderSequentially(funcp, logicClasses.m_final);
+    splitCheck(funcp);
+}
+
+class SenExprBuilder final {
+    // STATE
+    AstCFunc* const m_initp;  // The initialization function
+    AstScope* const m_scopeTopp;  // Top level scope
+    AstVarScope* const m_dpiExportTriggerp;  // The DPI export trigger variable
+
+    std::vector<AstNodeStmt*> m_updates;  // Update assignments
+
+    std::unordered_map<VNRef<AstNode>, AstVarScope*> m_prev;  // The 'previous value' signals
+    std::unordered_set<VNRef<AstNode>> m_hasUpdate;  // Whether the given sen expression already
+                                                     // has an update statement in m_updates
+
+    V3UniqueNames m_uniqueNames{"__Vtrigprev__expression"};  // For generating unique signal names
+
+    // METHODS
+    AstVarScope* getPrev(AstNode* currp) {
+        FileLine* const flp = currp->fileline();
+        const auto rdCurr = [=]() { return currp->cloneTree(false); };
+
+        // Create the 'previous value' variable
+        auto it = m_prev.find(*currp);
+        if (it == m_prev.end()) {
+            // For readability, use the scoped signal name if the trigger is a simple AstVarRef
+            string name;
+            if (AstVarRef* const refp = VN_CAST(currp, VarRef)) {
+                AstVarScope* vscp = refp->varScopep();
+                name = "__Vtrigrprev__" + vscp->scopep()->nameDotless() + "__"
+                       + vscp->varp()->name();
+            } else {
+                name = m_uniqueNames.get(currp);
+            }
+
+            AstVarScope* const prevp = m_scopeTopp->createTemp(name, currp->dtypep());
+            it = m_prev.emplace(*currp, prevp).first;
+
+            // Add the initializer init
+            AstNode* const initp = rdCurr();
+            m_initp->addStmtsp(
+                new AstAssign{flp, new AstVarRef{flp, prevp, VAccess::WRITE}, initp});
+        }
+
+        AstVarScope* const prevp = it->second;
+
+        // Add update if it does not exist yet in this round
+        if (m_hasUpdate.emplace(*currp).second) {
+            m_updates.push_back(
+                new AstAssign{flp, new AstVarRef{flp, prevp, VAccess::WRITE}, rdCurr()});
+        }
+
+        //
+        return prevp;
+    }
+
+    std::pair<AstNode*, bool> createTerm(AstSenItem* senItemp) {
+        if (senItemp->edgeType() == VEdgeType::ET_ILLEGAL) {
+            senItemp->v3warn(
+                E_UNSUPPORTED,
+                "Unsupported: Complicated event expression in sensitive activity list");
```

Comment:
Need test for this.

Reply:
Actually, ET_ILLEGAL should just go away as event expressions are now supported. I will fix the parser appropriately and add a test for it.

Reply:
Ok, arbitrary expressions would work with the scheduling implementation, but they are so hairy in the standard that I will punt this to a later patch as they need a lot of tests and validation.

Reply:
There is already a test for this (t_lint_unsup_mixed), which I tweak to hit this case, but no need for a message here as we can only reach it with --bbox-unsup, which turns of the unsupported message as well...